spec/services/course_progress_spec.rb

```diff
@@ -92,6 +92,10 @@
         expect(completion_time.to_s).to eq '2022-06-30 00:30:00 UTC'
       end
     end
+
+    it do
```

Comment:
This has been dropped into a block testing another method. Really it needs its own block, with its own contexts.

```ruby
describe '#completed_all_modules?' do
```

This is the spec I'd focus on. If it is fully covered here we don't need to lean on system specs as much.